application/apps/indexer/plugins_host/src/plugins_manager/mod.rs

```diff
@@ -156,4 +159,166 @@ impl PluginsManager {
 
         Ok(())
     }
+
+    /// Adds a plugin with the given directory path and the optional plugin type.
+    ///
+    /// * `plugin_dir_src`: Path of the plugin directory to be copied into chipmunk plugins directory.
+    /// * `plugin_type`: Type of the plugin, when not provided plugin type will be entered from plugin
+    ///   `WIT` signature in its binary file.
+    pub async fn add_plugin(
+        &mut self,
+        plugin_dir_src: PathBuf,
+        plugin_type: Option<PluginType>,
+    ) -> Result<(), PluginsManagerError> {
```

Comment:
I am not sure, we need install/uninstall functions to be async. Even if we need to leave it as async (because of some API inside), I'm not sure about `tokio::fs`. I just want to say: installation of plugin isn't something what we want to make in really async way. Even more, if at some time we would allow users, for example install multiple plugins, we would not do it in parallel but sequentially (because it is always safer). 

Well up to you, but I just don't see any needs to use async `tokio::fs` in that context.

Reply:
yes, that makes sense. I've missed that moment